kernel/core/utils/stringtools.cpp

```diff
@@ -1016,18 +1050,129 @@ bool isConvertible(const std::string& sStr, ConvertibleType type)
             || tolower(sStr.back()) == 'e')
             return false;
 
+        // NEW
+        bool inNum = false;
+        int numStartIdx = 0;
+        int lastSepIdx = -1;
+
+        // TODO is a starting decimal seperator valid ? If so we know what Format hast do be
+        int nFormat = NUM_NONE;
+        if(sStr[0] == '.')
+            nFormat |= NUM_DECIMAL_US;
+        else if (sStr[0] == ',')
+            nFormat |= NUM_DECIMAL_EU;
+
+        if(nFormat != NUM_NONE){
+            lastSepIdx = 0;
+            inNum = true;
+        } else if('0' <= sStr[0] && sStr[0] <= '9' )
+            inNum = true;
+
         // Regression fix introduced because NA is accepted as NaN
-        for (size_t i = 1; i < sStr.length()-1; i++)
+        size_t i = 1;
+        for (; i < sStr.length(); i++)
         {
-            if (sStr[i] == '-' || sStr[i] == '+')
-            {
-                if (tolower(sStr[i-1]) != 'e'
-                    && (isdigit(sStr[i-1]) && sStr.find_first_of("iI", i+1) == std::string::npos)
-                    && sStr[i-1] != ' ')
-                    return false;
+            //if(sStr[i] < '0' || '9' < sStr[i]) { // No seperator and no Decimal -> outside of Number
+            if(sStr[i] == ',' || sStr[i] == '.' || sStr[i] == ' ' || ('0' <= sStr[i] && sStr[i] <= '9' )) {
+                if(!inNum){
+                    inNum = true;
+                    numStartIdx = i;
+                }
+                if(sStr[i] == ',' || sStr[i] == '.' || sStr[i] == ' ' ) {

```

Comment:
This huge block appears to replicate a lookup table. If you could transform it in such a thing, that would greatly improve readability (and testability). I think, you're familiar with the concept of cyclomatic complexity. I counted a 17 for this block, which is beyond acceptable limits.

Consider the following possible solutions:

* Create a lookup table, where you can search for thousand seps and decimal seps and the table will return you the corresponding bitflags
* Create a map of all possible combinations (linear lookup table), if that is handleable, and use `std::map::find()` to look for the correct number format, e.g., `auto iter = numberFormats.find("xxx,xxx.xxx"); if (iter != numberFormats.end()) return iter->second`
* Create a simple structure, where you can set the found separator in their order and the structure will then solve the number format for you